### PR TITLE
Add BrandSelect component

### PR DESCRIPTION
**Pull request by phongtcao**
Add `BrandSelect` component to `wingman-fe`

It also comes along with the mocked component and story book

### CODE REVIEW

File: fe/lib/index.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ export { MockJobCategorySuggest } from './components/JobCategorySuggest/JobCateg
 
 export { LocationSuggest } from './components/LocationSuggest/LocationSuggest';
 export { MockLocationSuggest } from './components/LocationSuggest/LocationSuggest.mock';
+export { BrandSelect } from './components/BrandSelect/BrandSelect';
+export { MockBrandSelect } from './components/BrandSelect/BrandSelect.mock';
```
Comment (72636c) [inline fe/lib/index.ts:14]: Nit: can we move these up to their own section above `JobCategorySelect`?

File: fe/lib/components/BrandSelect/styles.css.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { style } from '@vanilla-extract/css';
+import { calc } from '@vanilla-extract/css-utils';
+import { vars } from 'braid-design-system/css';
+
+const coverImageProps = style({
+  borderTopLeftRadius: vars.borderRadius.large,
+  borderTopRightRadius: vars.borderRadius.large,
+  height: calc.multiply(vars.grid, 40),
```
Comment (72636c) [inline fe/lib/components/BrandSelect/styles.css.ts:8]: Q: are our brand tiles too large? The AdCentre uses much smaller ones.

File: fe/lib/components/BrandSelect/queries.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+`;
+
+export const ADVERTISEMENT_BRANDING = gql`
+  query AdvertisementBranding($id: String!) {
+    advertisementBranding(id: $id) {
+      ...advertisementBrandingFields
+    }
+  }
+  ${ADVERTISEMENT_BRANDING_FIELDS}
+`;
```
Comment (72636c) [inline fe/lib/components/BrandSelect/queries.ts:32]: Where is this query used?

File: fe/lib/components/BrandSelect/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { BrandSelect } from './BrandSelect';
```
Comment (72636c) [inline fe/lib/components/BrandSelect/index.ts:1]: I think we can delete this file

File: fe/lib/components/ConnectionPagination.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import {
```
Comment (72636c) [inline fe/lib/components/ConnectionPagination.tsx:1]: If we're not planning to export `ConnectionPagination` and `CopyableOid`, they should live under the `/private` subdirectory.

File: fe/lib/components/BrandSelect/BrandGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import { BrandTile } from './BrandTile';
+
+interface Props {
+  edges: AdvertisementBrandingEdgeFieldsFragment[];
+  selectedBrandId?: string;
+  onSelect?: (selectedBrand: AdvertisementBrandingFieldsFragment) => void;
+}
+
+export const BrandGrid = ({ edges, selectedBrandId, onSelect }: Props) => (
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandGrid.tsx:16]: I wonder if we should rename this component to `BrandTiles`

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+  const { data: categoriesData } = useQuery<JobCategoriesQuery>(
+    JOB_CATEGORIES,
+    {
+      ...(client && { client }),
+      variables: {
+        schemeId: 'seekANZ',
+      },
+    },
+  );
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:67]: Was this just for testing? Doesn't seem to belong here.

Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.tsx:67]: yeah, I tested it to see if it's mocked ok with categories, with the mocked apollo client, I should have cleaned it up

File: fe/lib/components/BrandSelect/queries.ts (hunk 3 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/queries.ts]: removed, it's not used in wingman

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        Brand <Secondary>(optional)</Secondary>
+      </Heading>
+
+      <PaginatedBrands
+        data={renderedData}
+        pageSize={PAGE_SIZE}
+        selectedBrandId={selectedBrandId}
+        onPagination={(args) => {
+          setVariables({ hirerId, ...args });
+        }}
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:97]:
```suggestion
        onPagination={(args) => setVariables({ ...args, hirerId })}
```

File: fe/lib/components/BrandSelect/BrandTile.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import { CopyableOid } from '../../private/CopyableOid';
+import {
+  AdvertisementBrandingFieldsFragment,
+  AdvertisementBrandingImage,
+} from '../../types/seekApi.graphql';
+
+import * as styles from './styles.css';
+
+const GREY_PIXEL =
+  'data:image/gif;base64,R0lGODlhAQABAIAAAMLCwgAAACH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==';
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandTile.tsx:13]: 1. Are there cases where we could get a brand without a logo?
2. Should this be unified with the chequerboard pattern used for cover images?

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        />
+      ) : null}
+    </Stack>
+  ) : (
+    <FieldMessage
+      id="noBrandsMesssage"
+      message="You have no brands to add to your job ad"
+      tone="neutral"
+    />
+  );
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:115]: This seems superfluous; `PaginatedBrands` already handles this case.

File: fe/lib/components/BrandSelect/BrandTitles.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+import { BrandTile } from './BrandTile';
+
+interface Props {
+  edges: AdvertisementBrandingEdgeFieldsFragment[];
+  selectedBrandId?: string;
+  onSelect?: (selectedBrand: AdvertisementBrandingFieldsFragment) => void;
+}
+
+export const BrandTitles = ({ edges, selectedBrandId, onSelect }: Props) => (
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandTitles.tsx:17]: Sorry for confusion:

```diff
- BrandTitles
+ BrandTiles
```

File: fe/lib/components/BrandSelect/styles.css.ts (hunk 2 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/styles.css.ts:8]: hey @72636c can you send me the AdCentre link which you refer to? Is it https://talent.seek.com.au/account/branding/?  I see the tiles there are  almost similar size

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+import { BrandSelect as Component } from './BrandSelect';
+import { MockBrandSelect } from './BrandSelect.mock';
+
+export default {
+  args: {
+    braidThemeName: defaultArgs.braidThemeName,
+    id: 'brands',
+    label: 'Brand',
+    message: 'undefined',
```
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:20]: yeah, I copied from CategorySelect, that I would need to review again to understand what they're doing to make them more specific to Branding

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+      control: { type: 'radio' },
+      mapping: { undefined, requiredValidation: 'Please select a brand.' },
+      options: ['undefined', 'requiredValidation'],
+    },
+    showStorybookAction: defaultArgTypes.showStorybookAction,
+    tone: defaultArgTypes.tone,
+  },
+  component: Component,
+
+  title: 'Job Posting/Brand/BrandSelect',
```
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: tbh, I'm quite confused with the terms, I thought we renamed it from AdvertisementBranding to Brand but seems like we refer to it as `Branding` in public documentation, don't we?

In AdCentre, the url is `branding` but the title is `Brands` @@
Screen Shot 2021-09-08 at 5 17 35 pm

File: fe/lib/components/BrandSelect/BrandTitles.tsx (hunk 11 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandTitles.tsx:17]: oh, sorry for the typo 😓

File: fe/lib/components/BrandSelect/styles.css.ts (hunk 2 above)
Comment (72636c) [reply fe/lib/components/BrandSelect/styles.css.ts:8]: Ah, I meant the widget that's part of the job posting flow.

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: Screen Shot 2021-09-08 at 5 26 18 pm

File: fe/lib/components/BrandSelect/styles.css.ts (hunk 2 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/styles.css.ts:8]: ah, I see

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: Not sure why I see a slightly different layout in AdCentre (while I'm in of job posting flow)

Screen Shot 2021-09-08 at 5 36 47 pm

[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: Oh wow, interesting. Maybe an A/B test, and I do like your screenshot more 🙂

[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: so... do you think we should keep it as `Brand` or change it to `Branding` as the general concept?

[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:38]: I think we should copy one of the two examples above, and it looks like our [UX guide](https://developer.seek.com/use-cases/job-posting/user-experience-guide) uses _Branding_.

File: fe/lib/components/BrandSelect/BrandTile.tsx (hunk 9 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandTile.tsx]: Added a `No logo` case but realised that I cannot create a Brand without a logo in AdCentre, so just simplify no need  `No Logo` case

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+    AdvertisementBrandingsQuery,
+    AdvertisementBrandingsQueryVariables
+  >(ADVERTISEMENT_BRANDINGS, {
+    ...(client && { client }),
+    variables,
+  });
+
+  const [selectedBrandId, setSelectedBrandId] = useState(initialBrandId);
+
+  useEffect(() => {
```
Comment (phongtcao) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:57]: useEffect here to take `initialBrandId` effect in story book. 
wondering whether we have the initialBrandId changed in real scenario or not? to consider if it's necessary

File: fe/lib/components/BrandSelect/BrandTile.tsx (hunk 9 above)
Comment (etaoins) [reply fe/lib/components/BrandSelect/BrandTile.tsx]: Sorry for jumping on this late, but that’s correct. It’s also documented on the Developer Site:
> A mandatory logo displayed on the hirer’s job ads and on SEEK’s native apply form. This is represented as an AdvertisementBrandingImage  with a type code of OriginalLogo.

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx]: I renamed it to Branding so it aligns with Indirect space for now,  later we can review to change for consistency elsewhere

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+    endCursor: endCursorPaginatedBrandsLastPage,
+  },
+  edges: [
+    {
+      node: {
+        id: {
+          value:
+            'global:advertisementBranding:hirerBranding:qVe8DCsXJP9zNHsmcmEyL',
+        },
+        name: 'Spotlight - test',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: 1. Can we upload some more neutral branding examples? They could just be some public domain patterned/gradient backgrounds and such. We shouldn't include the brands of real organizations or Ryanair here.

File: fe/lib/components/BrandSelect/PaginatedBrands.tsx [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  onSelect,
+}: Props) => {
+  const brands = data?.advertisementBrandings?.edges || [];
+  const hasAtLeastOneBrand = data && brands.length > 0;
+
+  if (!hasAtLeastOneBrand) {
+    return (
+      <Notice tone="info">
+        <Text>
+          You have no brand. Create a new one in the{' '}
```
Comment (72636c) [inline fe/lib/components/BrandSelect/PaginatedBrands.tsx:38]:
```suggestion
          You have no brands. Create a new one in the{' '}
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts (hunk 15 above)
Comment (etaoins) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: We should feed that back in to public test as well

Comment (etaoins) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: Do you think it's safe to use SEEK's logo? The brands in public test use some of the generic SEEK branding:
<https://image-service-cdn.seek.com.au/27ecca452d06d960b596aa40380f0d06c56b6ba8>
<https://image-service-cdn.seek.com.au/7a2b5aa53d67c7c6d0eeae95bfa56d9b2f5ac228>

Comment (phongtcao) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: @etaoins where can I get some more of those images as your example? or do I need to upload some of them as the sample brands then get the url from there?

File: fe/lib/components/BrandSelect/styles.css.ts (hunk 2 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/styles.css.ts:8]: @72636c I've checked to resize the tiles, but it's responsive to the container, any suggestions to style it? and wonder how it looks when used in Ryanair Position Posting form

Comment (72636c) [reply fe/lib/components/BrandSelect/styles.css.ts:8]: All good Phong, I'm happy to leave as-is and we can always tweak it on later UX review. This is a big effort already without me scope creeping 😅

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts (hunk 15 above)
Comment (etaoins) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: I think they all initially came from the Ryanair test hirer. You should be able use Web Inspector to get the out of Ryanair, or run `yarn printToken:prod` and run `advertisementBrandings` yourself. The issue is avoiding the third party & joke brands in Ryanair.

Comment (phongtcao) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: yeah, I see what you meant, I also actually copied the responses from queries in ryanair to make them as mocked data, but looks like it includes some information which is not good for mocked data. 

What I meant is that if I want to use some other covers for my mocked data then where I can get them more. @72636c gave me the link to generate some more in [public test data](https://github.com/SEEK-Jobs/indie-public-test-data/blob/da9c02fc0b7dcb8952fa3df1358310f13e131065/mock-http-server/httpMocks/bx-resource-api/brands/mockBrands.js#L69), I'll look into it to generate some more to demonstrate pagination

Comment (etaoins) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts:289]: Ah yeah, I think if you create them on `talent.seek.com.au` they'll get uploaded to the CDN permanently.

I wonder if you could use some of these as cover images (they're free for commercial use):
<https://unsplash.com>

Comment (phongtcao) [reply fe/lib/components/BrandSelect/__fixtures__/brands.ts]: Update mocked brands with Seek logo and some random free covers. I'll feed back some brands to public test data later

File: fe/lib/components/BrandSelect/README.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
+# BrandSelect
+
+A BrandSelect component that provides a full list of advertisement brandings associated with the specified hirer.
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:3]:
```suggestion
A BrandSelect component that provides a paginated selection of brands associated with the specified hirer.
```

File: fe/lib/components/BrandSelect/README.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+### Properties
+
+Required
+
+- `hirerId`: The identifier for the [HiringOrganization](https://developer.seek.com/schema/#/named-type/HiringOrganization).
+
+Optional:
+
+- `client`: An `ApolloClient` instance. By default BrandSelect uses the client passed down via context, but a different client can be passed in.
+- `onChange`: Callback function that is supplied a [Brand identifier](https://developer.seek.com/schema/#/named-type/AdvertisementBranding/field/id) on brand selection.
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:25]:
```suggestion
- `onChange`: Callback function that is supplied an [AdvertisementBranding identifier](https://developer.seek.com/schema/#/named-type/AdvertisementBranding/field/id) on brand selection.
```

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+export interface BrandSelectProps {
+  client?: ApolloClient<unknown>;
+  hirerId: string;
+  errorMessage?: string;
+  initialBrandId?: string;
+  pageSize?: number;
+  /**
+   * Type definition from the `Controller` component that's provided by `react-hook-form`.
+   */
+  onChange: (...event: any[]) => void;
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:30]: 1. What's going on with the type here? This is pretty unusable for consumers.
1. All the other components seem to name this `onSelect`; we should align these one way or the other.

File: fe/lib/components/BrandSelect/README.md [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+
+Extends:
+
+- You may pass through various HTML input field props including: `id`, `name`, `label`
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:29]: This does not match the implementation.

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+import { BraidStorybookProvider } from '../../storybook/decorators';
+
+import { BrandSelect as Component } from './BrandSelect';
+import { MockBrandSelect } from './BrandSelect.mock';
+
+export default {
+  args: {
+    braidThemeName: defaultArgs.braidThemeName,
+    id: 'brands',
+    label: 'Brand',
```
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:19]: Removed

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+      'global:advertisementBranding:hirerBranding:4pkLmqYhoSxSKmfcKMbDG6',
+    pageSize: 4,
+  },
+  argTypes: {
+    braidThemeName: defaultArgTypes.braidThemeName,
+    showStorybookAction: defaultArgTypes.showStorybookAction,
+    mockHasNextPage: {
+      name: 'Has next brands page',
+      control: { type: 'boolean' },
+    },
```
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:31]: @72636c This is only for the storyBook to simulate pagination, what should we do in this case as it should not be part of `BrandSelectProps`?

[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: Oh my bad, I missed that this is a prop on `MockBrandSelect`. Is it possible for our mock resolver to be dynamic on the actual variables supplied by `BrandSelect`?

[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: yes, I already handled some scenarios in the mocked resolver for some scenarios in pagination. 

This setting I just try to demonstrate the scenario only one page in the first query, no Prev/ Next link
Screen Shot 2021-09-13 at 11 28 53 am

[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: Maybe it's unnecessary and overcomplicated, I'll remove this option. Just support pagination with a pageSize configuration

[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: Ah gotcha! Yeah I think a large `pageSize` should do the trick here ✨

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+import { BrandSelect as Component } from './BrandSelect';
+import { MockBrandSelect } from './BrandSelect.mock';
+
+export default {
+  args: {
+    braidThemeName: defaultArgs.braidThemeName,
+    id: 'brands',
+    label: 'Brand',
+    hirerId: 'seekAnz:organization:seek:kLDHs7W7',
```
[the remark this answers is not in the document]
Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:18]: 🤔 did this get reverted again?

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+      'global:advertisementBranding:hirerBranding:4pkLmqYhoSxSKmfcKMbDG6',
+    pageSize: 4,
+  },
+  argTypes: {
+    braidThemeName: defaultArgTypes.braidThemeName,
+    showStorybookAction: defaultArgTypes.showStorybookAction,
+    pageSize: {
+      control: { type: 'radio' },
+      options: [4, 6],
+    },
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: Do we need to explicitly mark this up, or can Storybook infer a numeric control?

```suggestion
```

File: fe/lib/components/BrandSelect/README.md [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+    <form action="" method="post" onSubmit={handleSubmit(handleFormSubmit)}>
+      <Controller
+        control={control}
+        name="brandingId"
+        render={({ field }) => (
+          <BrandSelect
+            {...field}
+            client={client}
+            id="brandingId"
+            label="Brandings"
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:104]: Are these valid?

File: fe/lib/components/BrandSelect/README.md [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+
+// client set-up as per the previous example
+
+const BrandingForm = () => {
+  const [brandingId, setBrandingId] = useState();
+  return (
+    <div>
+      <BrandingForm
+        schemeId="seekAnz"
+        client={client}
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:70]: 1. Should this be `BrandSelect`?
2. `schemeId` looks like copypasta
3. Can we rely on an implicit `client` here as per _Default usage_? That's the typical case and we can strip this example down to focus solely on `onSelect`.

If this is getting overly tedious, I'd rather drop the usage examples than have inaccurate ones. We can investigate having Storybook-powered examples in future.

File: fe/lib/components/BrandSelect/README.md [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+
+## Installation
+
+```shell
+yarn add wingman-fe
+```
+
+## Brand Select Widget
+
+The BrandSelect widget abstracts the `advertisementBrandings` query on SEEK API and provides a select input that returns a full list of advertisement brandings associated with the specified hirer.
```
Comment (72636c) [inline fe/lib/components/BrandSelect/README.md:13]: "Full list" could be misleading when it implements pagination

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 28]
```diff
@@ -0,0 +1,254 @@
...
+  'global:advertisementBrandingCursor:indirectBranding:end3';
+
+const mockBrands = [
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:MjHScMntyZKfPPKv37EqAn',
+      },
+      name: ' Seek logo - no cover image',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:21]:
```suggestion
      name: 'SEEK logo - No cover image',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 29]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:4pkLmqYhoSxSKmfcKMbDG6',
+      },
+      name: ' Seek logo - macbook (green)',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:40]:
```suggestion
      name: 'SEEK logo - MacBook (green)',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 30]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:CNEHkUP51vvPyYuBwfQw7u',
+      },
+      name: 'Seek logo - notebook',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:59]:
```suggestion
      name: 'SEEK logo - Notebook',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 31]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:MD2FWYzmVd4p5Bs1pFEddd',
+      },
+      name: ' Seek logo - laptop',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:78]:
```suggestion
      name: 'SEEK logo - Laptop',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 32]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:4YsFiFRekSvuKjLzRUtmZH',
+      },
+      name: 'Seek logo - pens',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:97]:
```suggestion
      name: 'SEEK logo - Pens',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 33]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:2P8WhewQF4dnZNYRKMDv1r',
+      },
+      name: 'Seek logo - blue cover',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:116]:
```suggestion
      name: 'SEEK logo - Blue cover',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 34]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:VQ9aAzkhogAMSw3AF31stC',
+      },
+      name: 'Seek logo blue - backpack',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:135]:
```suggestion
      name: 'SEEK blue logo - Backpack',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 35]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:D557J8rpyMKpP9DxrWg2s',
+      },
+      name: 'Seek logo blue - sunset',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:154]:
```suggestion
      name: 'SEEK blue logo - Sunset',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 36]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:qVe8DCsXJP9zNHsmcmEyL',
+      },
+      name: 'Seek logo - Town',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:173]:
```suggestion
      name: 'SEEK logo - Town',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 37]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:7yQLDLdqz9phfe9j1GHDgn',
+      },
+      name: 'Seek logo - Beach',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:192]:
```suggestion
      name: 'SEEK logo - Beach',
```

File: fe/lib/components/BrandSelect/__fixtures__/brands.ts [hunk 38]
```diff
@@ -0,0 +1,254 @@
...
+      ],
+    },
+  },
+  {
+    node: {
+      id: {
+        value:
+          'global:advertisementBranding:hirerBranding:33Y863h6sCnrCFk97w9bvg',
+      },
+      name: 'Seek logo - Flower',
```
Comment (72636c) [inline fe/lib/components/BrandSelect/__fixtures__/brands.ts:211]:
```suggestion
      name: 'SEEK logo - Flower',
```

File: fe/lib/components/BrandSelect/PaginatedBrands.tsx [hunk 39]
```diff
@@ -0,0 +1,63 @@
...
+
+export const PaginatedBrands = ({
+  data,
+  pageSize,
+  selectedBrandId,
+  onPagination,
+  onSelect,
+}: Props) => {
+  const brands = data?.advertisementBrandings?.edges || [];
+  const hasAtLeastOneBrand = data && brands.length > 0;
```
Comment (72636c) [inline fe/lib/components/BrandSelect/PaginatedBrands.tsx:32]:
```suggestion
  const hasAtLeastOneBrand = data?.advertisementBrandings?.edges.length;
```

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 40]
```diff
@@ -0,0 +1,109 @@
...
+      <Inline space="small">
+        <Text>Loading brands</Text>
+        <Loader size="xsmall" />
+      </Inline>
+    );
+  }
+
+  return (
+    <Stack space="medium">
+      <Heading level="4" weight={'regular'}>
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:86]:
```suggestion
      <Heading level="4">
```

File: fe/lib/components/BrandSelect/BrandSelect.tsx [hunk 41]
```diff
@@ -0,0 +1,109 @@
...
+import { ADVERTISEMENT_BRANDINGS } from './queries';
+
+export interface BrandSelectProps {
+  client?: ApolloClient<unknown>;
+  hirerId: string;
+  initialBrandId?: string;
+  pageSize?: number;
+  /**
+   * Type definition from the `Controller` component that's provided by `react-hook-form`.
+   */
```
Comment (72636c) [inline fe/lib/components/BrandSelect/BrandSelect.tsx:28]:
```suggestion
```

**Review by 72636c — APPROVED** [review]

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:18]: oh damn, sorry

File: fe/lib/components/BrandSelect/README.md (hunk 25 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/README.md:104]: No 😞

File: fe/lib/components/BrandSelect/BrandSelect.stories.tsx (hunk 24 above)
Comment (phongtcao) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: I got what you meant, the reason I only give it 2 options in pageSize that the mocked resolver handles for with max 3 pages on the same mocked dataset of 11 brands. If we allow to types pageSize, it doesn't know to handle dynamically in pages navigation. 

Any suggestion in the mock resolver more dynamically?

Comment (72636c) [reply fe/lib/components/BrandSelect/BrandSelect.stories.tsx:29]: Ah no stress. Feel free to merge as is, I'll take a look later.

**Pull request merged**